lib/spack/spack/cmd/edit.py

```diff
@@ -112,9 +117,30 @@ def edit(parser, args):
     if args.path:
         path = args.path
         if name:
-            path = join_path(path, name + ".py")
-            if not os.path.exists(path):
-                tty.die("No command for '{0}' was found.".format(name))
+            if '.' in name:
+                # User provided suffix. Just use join it to the path
+                path = join_path(path, name)
```

Comment:
I've been trying to phase out the use of `join_path` in core. It's almost identical to `os.path.join`, except that it converts arguments to strings.